Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,682 @@
+        locationManager.stopUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.stopUpdatingHeading()
+        }
+    }
+
+    /// Start the locationManager with denied access
+    fileprivate func startWithAccessDenied() {
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Start the locationManager with authorized access
+    fileprivate func startWithAccessAuthorized() {
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Potential notification to the user of an error starting device tracking
+    ///
+    /// - Parameter error: error that ocurred when starting tracking
+    fileprivate func didStartOrFailWithError(_ error: Error?) {
+        if !notifiedStartOrFailure, let error = error {
+            // TODO: present error to user...
+            print("didStartOrFailWithError: \(String(reflecting:error))")
+        }
+        
+        notifiedStartOrFailure = true;
+    }
+    
+    /// Handle a change in authorization status to "denied"
+    fileprivate func handleAuthStatusChangedAccessDenied() {
+        // auth status changed to denied
+        stopUpdatingLocationAndHeading()
+        
+        // we were waiting for user prompt to come back, so notify
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Handle a change in authorization status to "authorized"
+    fileprivate func handleAuthStatusChangedAccessAuthorized() {
+        // auth status changed to authorized
+        // we were waiting for status to come in to start the datasource
+        // now that we have authorization - start it
+        didStartOrFailWithError(nil)
+        
+        // need to start location manager updates
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Called when device orientation changes
+    ///
+    /// - Parameter notification: the notification
+    @objc func orientationChanged(notification: Notification?) {
+        // handle rotation here
+        switch UIApplication.shared.statusBarOrientation {
+        case .landscapeLeft:
+            orientationQuat = simd_quaternion(0, 0, 1.0, 0);
+        case .landscapeRight:
+            orientationQuat = simd_quaternion(0, 0, 0, 1.0);
+        case .portrait:
+            orientationQuat = simd_quaternion(0, 0, sqrt(0.5), sqrt(0.5));
+        case .portraitUpsideDown:
+            orientationQuat = simd_quaternion(0, 0, -sqrt(0.5), sqrt(0.5));
+        default:
+            break
+        }
+    }
+    
+    // MARK: Errors
+    
+    /// Error used when ARKit is not supported on the current device
+    ///
+    /// - Returns: error stating ARKit not supported
+    fileprivate class func notSupportedError() -> NSError {
+        let userInfo = [NSLocalizedDescriptionKey : "The device does not support ARKit functionality."]
+        return NSError(domain: AGSErrorDomain, code: 0, userInfo: userInfo)
```

Comment:
is there an `AGSErrorCode` value that works for this? `AGSErrorCodeCommonIllegalState`?

Reply:
Does `AGSErrorCodeCommonPropertyNotSupported` work?

Reply:
How about `ARErrorCodeUnsupportedConfiguration`?

Reply:
seems fine